client/src/main/java/org/apache/celeborn/client/read/PartitionReader.java

```diff
@@ -31,4 +33,8 @@ public interface PartitionReader {
   void close();
 
   PartitionLocation getLocation();
+
+  Optional<PartitionReaderCheckpointMetadata> getPartitionReaderCheckpointMetadata();
```

Comment:
You can add a chunk index in the method of `next` to make sure that you can record the chunk that has been fully consumed.